### PR TITLE
guard against the possibility of no links being provided

### PR DESCRIPTION
**Pull request by alfredodeza**
Closes https://github.com/anchore/scan-action/issues/68

/cc @jonico

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -221,6 +221,12 @@ function grype_render_rules(vulnerabilities) {
         let ruleID = "ANCHOREVULN_"+v.vulnerability.id+"_"+v.artifact.type+"_"+v.artifact.name+"_"+v.artifact.version;
         if (!ruleIDs.includes(ruleID)) {
           ruleIDs.push(ruleID);
+          // Entirely possible to not have any links whatsoever
+          var link = v.vulnerability.id;
```
Comment (luhring) [inline index.js:225]: Not blocking, but in general `var` should never be used, and it's safe to use `let` in its place

File: index.js [hunk 2]
```diff
@@ -221,6 +221,12 @@ function grype_render_rules(vulnerabilities) {
...
+          if ("links" in v.vulnerability) {
```
Comment (luhring) [inline index.js:226]: I haven't seen this style of check before, but that doesn't mean it's wrong. I usually see checks for property's existence in the form of `if (typeof v.vulnerability.links !== "undefined") ...`

File: index.js [hunk 3]
```diff
@@ -240,11 +246,11 @@ function grype_render_rules(vulnerabilities) {
                     "Location: "+v.artifact.locations[0].path+"\n"+
                     //"Data Namespace: "+v.vulnerability.matched_by.matcher +"\n"+
                     "Data Namespace: "+ "unknown" + "\n"+
-                    "Link: ["+v.vulnerability.id+"]("+v.vulnerability.links[0]+")",
+                    `Link: ${link}`,
                     "markdown": "**Vulnerability "+v.vulnerability.id+"**\n"+
                     "| Severity | Package | Version | Fix Version | Type | Location | Data Namespace | Link |\n"+
                     "| --- | --- | --- | --- | --- | --- | --- | --- |\n"+
-                    "|"+v.vulnerability.severity+"|"+v.artifact.name+"|"+v.artifact.version+"|"+"unknown"+"|"+v.artifact.type+"|"+v.artifact.locations[0].path+"|"+"unknown"+"|["+v.vulnerability.id+"]("+v.vulnerability.links[0]+")|\n"
+                    "|"+v.vulnerability.severity+"|"+v.artifact.name+"|"+v.artifact.version+"|"+"unknown"+"|"+v.artifact.type+"|"+v.artifact.locations[0].path+"|"+"unknown"+"|"+link+"|\n"
```
Comment (luhring) [inline index.js:253]: This whole line might read easier as a template, and with values separated out. For example:

```javascript
const { severity } = v.vulnerability;
const { name, version, type, locations } = v.artifact;
const { path } = locations[0];

// ...

`|${severity}|${name}|${version}|unknown|${type}|${path}|unknown|${link}|\n`
```

**Review by luhring — APPROVED** [review]

File: index.js (hunk 3 above)
Comment (alfredodeza) [reply index.js:253]: I agree. I've started using templates as I carve out changes. This whole file needs to use more templates. Do note that I have captured that feedback and created https://github.com/anchore/scan-action/issues/51 to track it.

File: index.js (hunk 1 above)
Comment (alfredodeza) [reply index.js]: I slipped on this one and have updated it to fix it. I ran tests locally against the reproducer, and it passed

**Pull request merged**